model/search_params_test.go

```diff
@@ -12,924 +12,1500 @@ import (
+				},
+			},
+		},
+		{
+			Name:  "string is multiple words containing a : and multiple # prefixes",
+			Input: "#apple #banana from:chan",
+			Words: []searchWord{
+				{
+					value:   "#apple",
+					exclude: false,
+				},
+				{
+					value:   "#banana",
+					exclude: false,
+				},
+			},
+			Flags: []flag{
+				{
+					name:  "from",
+					value: "chan",
+				},
+			},
+		},
+		{
+			Name:  "string is multiple words containing a : with a single - and multiple # prefixes",
+			Input: "-#apple #banana from:chan",
+			Words: []searchWord{
+				{
+					value:   "#apple",
+					exclude: true,
+				},
+				{
+					value:   "#banana",
+					exclude: false,
+				},
+			},
+			Flags: []flag{
+				{
+					name:  "from",
+					value: "chan",
+				},
+			},
+		},
+		{
+			Name:  "string is multiple words containing a : prefixed with - and multiple # prefixes",
+			Input: "#apple #banana -from:chan",
+			Words: []searchWord{
+				{
+					value:   "#apple",
+					exclude: false,
+				},
+				{
+					value:   "#banana",
+					exclude: false,
+				},
+			},
+			Flags: []flag{
+				{
+					name:    "from",
+					value:   "chan",
+					exclude: true,
+				},
+			},
+		},
+		{
+			Name:  "string is multiple words containing a : prefixed with multiple - and multiple # prefixes",
+			Input: "-#apple -#banana -from:chan",
+			Words: []searchWord{
+				{
+					value:   "#apple",
+					exclude: true,
+				},
+				{
+					value:   "#banana",
+					exclude: true,
+				},
+			},
+			Flags: []flag{
+				{
+					name:    "from",
+					value:   "chan",
+					exclude: true,
+				},
+			},
+		},
+		{
+			Name:  "string is multiple words containing a : with a space",
+			Input: "apple banana from: chan",
+			Words: []searchWord{
+				{
+					value:   "apple",
+					exclude: false,
+				},
+				{
+					value:   "banana",
+					exclude: false,
+				},
+			},
+			Flags: []flag{
+				{
+					name:  "from",
+					value: "chan",
+				},
+			},
+		},
+		{
+			Name:  "string is multiple words containing a in: with a space",
+			Input: "apple banana in: chan",
+			Words: []searchWord{
+				{
+					value:   "apple",
+					exclude: false,
+				},
+				{
+					value:   "banana",
+					exclude: false,
+				},
+			},
+			Flags: []flag{
+				{
+					name:  "in",
+					value: "chan",
+				},
+			},
+		},
+		{
+			Name:  "string is multiple words containing a channel: with a space",
+			Input: "apple banana channel: chan",
+			Words: []searchWord{
+				{
+					value:   "apple",
+					exclude: false,
+				},
+				{
+					value:   "banana",
+					exclude: false,
+				},
+			},
+			Flags: []flag{
+				{
+					name:  "channel",
+					value: "chan",
+				},
+			},
+		},
+		{
+			Name:  "string is only fruit: with a space",
+			Input: "fruit: cherry",
+			Words: []searchWord{
+				{
+					value:   "fruit",
+					exclude: false,
+				},
+				{
+					value:   "cherry",
+					exclude: false,
+				},
+			},
+			Flags: []flag{},
+		},
+		{
+			Name:  "string is a single word with a : postfix",
```

Comment:
I would call it `string with the a flag but without the value for that flag should be threaded as a word.`